src/components/SingleSelect.js

```diff
@@ -0,0 +1,47 @@
+import React, { useCallback, useMemo } from 'react'
+import propTypes from '@dhis2/prop-types'
+import { SingleSelectField, SingleSelectOption } from '@dhis2/ui-core'
+
+import { FieldAdapter } from './FieldAdapter.js'
+
+const useChangeHandler = onChange =>
+    useCallback(
+        ({ selected }) => {
+            onChange({ value: selected.value }, null)
+        },
+        [onChange]
+    )
+
+const useSelectedOption = (options, value) =>
+    useMemo(() => options.find(o => o.value === value) || {}, [options, value])
+
+const SingleSelectComponent = ({ options, value, onChange, ...rest }) => {
+    const handleChange = useChangeHandler(onChange)
+    const selectedOption = useSelectedOption(options, value)
```

Comment:
Here you're assuming that `options` is an array, but the prop is not required, so it could be `null` or `undefined` (just like in the multi select)